aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -0,0 +1,423 @@
+import software.amazon.awssdk.services.glue.model.GetDatabaseRequest;
+import software.amazon.awssdk.services.glue.model.GetDatabaseResponse;
+import software.amazon.awssdk.services.glue.model.GetDatabasesRequest;
+import software.amazon.awssdk.services.glue.model.GetDatabasesResponse;
+import software.amazon.awssdk.services.glue.model.GetTableRequest;
+import software.amazon.awssdk.services.glue.model.GetTableResponse;
+import software.amazon.awssdk.services.glue.model.GetTablesRequest;
+import software.amazon.awssdk.services.glue.model.GetTablesResponse;
+import software.amazon.awssdk.services.glue.model.InvalidInputException;
+import software.amazon.awssdk.services.glue.model.Table;
+import software.amazon.awssdk.services.glue.model.TableInput;
+import software.amazon.awssdk.services.glue.model.UpdateDatabaseRequest;
+
+public class GlueCatalog extends BaseMetastoreCatalog implements Closeable, SupportsNamespaces, Configurable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GlueCatalog.class);
+
+  public static final String CONFIG_GLUE_CATALOG_ID = "glue-catalog-id";
+  public static final String CONFIG_SKIP_ARCHIVE = "skip-archive";
+  public static final boolean CONFIG_SKIP_ARCHIVE_DEFAULT = false;
+
+  private final GlueClient glue;
+  private Configuration hadoopConf;
+  private String catalogId;
+  private String catalogName;
+  private String warehousePath;
+  private boolean skipArchive;
+  private FileIO fileIO;
+
+  /**
+   * No-arg constructor to load the catalog dynamically.
+   * <p>
+   * Only the AWS Glue client is initialized.
+   * Other fields must be initialized by calling {@link GlueCatalog#initialize(String, Map)} later.
+   */
+  public GlueCatalog() {
+    this.glue = AwsClientUtil.defaultGlueClient();
+  }
+
+  /**
+   * All-arg constructor to directly initialize the catalog.
+   * <p>
+   * All the fields are directly initialized.
+   * There is no need to call {@link GlueCatalog#initialize(String, Map)} later.
+   *
+   * @param glue Glue client
+   * @param catalogName a custom name for the catalog, usually obtained from a Spark or Flink session.
+   * @param catalogId The ID of the Glue Data Catalog where the tables reside.
+   *                  If none is provided, the AWS account ID is used by default.
+   * @param warehousePath warehouse path, such as s3://my-iceberg-bucket/metadata_prod
+   * @param skipArchive if Glue should skip archiving an old table version when creating a new version in a commit.
+   *                    By default Glue archives all old table versions after an UpdateTable call.
+   *                    but Glue has a default max number of archived table versions (can be increased).
+   *                    So for streaming use case with lots of commits, it is recommended to set this value to true.
+   * @param fileIO file IO to use.
+   */
+  public GlueCatalog(GlueClient glue, String catalogName, String catalogId, String warehousePath,
```

Comment:
Does this need to be public? It makes sense to have it for tests, but I would probably avoid public constructors that we need to support later.

Reply:
Since all this does is call `initialize`, maybe we don't even need it for tests. That method could be package-private.

Reply:
I will just have a package private constructor taking the glue client, and always call the initialize method then, which should simply the situation here a lot.